.github/workflows/CI.yml

```diff
@@ -106,9 +118,9 @@ jobs:
         run: python -m pytest --cov=umpost --cov-report=xml -s test
 
       - name: Upload code coverage
-        uses: codecov/codecov-action@v4
         # Only upload once for the installed version
-        if: matrix.python-version == ${{ env.PY_VERSION }}
+        if: ${{ matrix.python-version == env.PY_VERSION }}
+        uses: codecov/codecov-action@b9fd7d16f6d7d1b5d2bec1a2887e65ceed900238 #v4.6.0
         with:
           token: ${{ secrets.codecov_token }}
```

Comment:
I don't have access to the Repo settings, but this action throws a warning and does not upload the coverage file, which means the `secrets.codecov_token` might be expired or not set.